crates/rethnet_evm/src/state/overrides.rs

```diff
@@ -0,0 +1,178 @@
+use rethnet_eth::{account::KECCAK_EMPTY, Address, B256, U256};
+use revm::{
+    db::StateRef,
+    primitives::{AccountInfo, Bytecode, HashMap},
+};
+
+/// Type representing either a diff or full set of overrides for storage information.
+#[derive(Clone, Debug)]
+pub enum StorageOverride {
+    /// A diff of storage overrides.
+    Diff(HashMap<U256, U256>),
+    /// A full set of storage overrides.
+    Full(HashMap<U256, U256>),
+}
+
+/// Values for overriding account information.
+#[derive(Clone, Debug)]
+pub struct AccountOverride {
+    /// Account balance override.
+    pub balance: Option<U256>,
+    /// Account nonce override.
+    pub nonce: Option<u64>,
+    /// Account code override.
+    pub code: Option<Bytecode>,
+    /// Account storage override.
+    pub storage: Option<StorageOverride>,
+}
+
+impl AccountOverride {
+    /// Overrides the provided original account information.
+    pub fn override_info(&self, original: Option<AccountInfo>) -> Option<AccountInfo> {
```

Comment:
I would try to change the implementation of this method to make it a compile error if a new property is added to `AccountInfo` to prevent potential future bugs. One option would be to construct the `AccountInfo` explicitly in the return or to use destructuring to access the fields of the `AccountInfo` earlier.

Reply:
Good idea. Thanks!

Addressed by https://github.com/NomicFoundation/hardhat/pull/4463/commits/f5165bdcd8f9085c877aa74c45aa045b0e995826